### PR TITLE
[ETC] Endpoints fo create a user and enroll  a user

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Realise endpoints fo create a user and enroll a user

**Youtrack:** https://youtrack.raccoongang.com/issue/ETC-11

### CODE REVIEW

File: lms/djangoapps/etc_api/views.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            user.is_active = True
+            user.first_name = request.data.get('prename')
+            user.last_name = request.data.get('surname')
+            user.save()
+            self.send_activation_email(request)
+        except ValidationError as err:
+            # Should only get non-field errors from this function
+            assert NON_FIELD_ERRORS not in err.message_dict
+            # Only return first error for each field
+            return Response(data={"user_message": "Wrong parameters on user creation"}, status=400)
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: Which parameter is wrong? How can we get this from the error msg?

File: lms/djangoapps/etc_api/views.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            data['password'] = uuid4().hex
+            user = create_account_with_params(request, data)
+            user.is_active = True
+            user.first_name = request.data.get('prename')
+            user.last_name = request.data.get('surname')
+            user.save()
+            self.send_activation_email(request)
+        except ValidationError as err:
+            # Should only get non-field errors from this function
+            assert NON_FIELD_ERRORS not in err.message_dict
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: Please use `raise`  instead of `assert`

File: lms/djangoapps/etc_api/views.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = ApiKeyHeaderPermission,
+
+    def post(self, request):
+        """
+        Enable or disable user by user id.
+        """
+        data = request.data
+        try:
+            user = User.objects.get(id=data['user_id'])
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: What if required parameters are absent in the request?

File: lms/djangoapps/etc_api/views.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            user.is_active = string_to_boolean(data['is_active'])
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            user.save()
+        except  User.DoesNotExist:
+            return Response(data={"user_message": "Wrong id User does not exist"}, status=400)
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: s/Wrong id/Wrong user_id./

File: lms/djangoapps/etc_api/views.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    def post(self, request):
+        """
+        Enrolling user on the course with the specified mod.
+
+        Create or update enroll user on the course. Can deactivate enroll
+        or activate enroll with use 'is_active' param.
+        """
+        data = request.data
+        try:
+            user = User.objects.get(id=data['user_id'])
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        except  User.DoesNotExist:
+            return Response(data={"user_message": "Wrong id User does not exist"}, status=400)
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        course_id = SlashSeparatedCourseKey.from_deprecated_string(data['course_id'])
+        enrollment_obj, __ = CourseEnrollment.objects.update_or_create(
+            user=user,
+            course_id=course_id,
+            defaults={
+                'mode': data['mode'],
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+                'is_active': string_to_boolean(data['is_active'])
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+            }
+        )
+        return Response(data={'enrollment_id': enrollment_obj.id}, status=status.HTTP_200_OK)
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: Please add trailing empty line

**Comment by AndreyLykhoman** [comment]
I fixed issues in this PR: https://github.com/raccoongang/edx-platform/pull/572

File: lms/djangoapps/etc_api/views.py (hunk 2 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: I deleted this part.

File: lms/djangoapps/etc_api/views.py (hunk 1 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: I rewrote the message.

File: lms/djangoapps/etc_api/views.py (hunk 3 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: I added code for checking user_id.

File: lms/djangoapps/etc_api/views.py (hunk 4 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: I used 'get()' method and I rewrote method 'string_to_boolean' to work with None.

File: lms/djangoapps/etc_api/views.py (hunk 5 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: done

File: lms/djangoapps/etc_api/views.py (hunk 6 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: done

File: lms/djangoapps/etc_api/views.py (hunk 7 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: done

File: lms/djangoapps/etc_api/views.py (hunk 8 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: done

File: lms/djangoapps/etc_api/views.py (hunk 9 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: done

File: lms/djangoapps/etc_api/views.py (hunk 10 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: Added.

**Pull request merged**